File: exercises/C/chapter_4 - Strings/1_count_digits.c
// 1. Count digits in a string

// Design a Python program that will ask the user to enter a string.
// The program will count the sum of all digits in the string.
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#define BUFFER_LEN 1000

int main(int argc, char const *argv[])
{
    char string[BUFFER_LEN];
    int digit_count = 0;
    int sum = 0;

    printf("Enter a string: ");
    fgets(string, BUFFER_LEN, stdin);
    string[strcspn(string, "\n")] = '\0'; // Remove trailing newline

    for (int i = 0, len = strlen(string); i < len; i++)
    {
        if (string[i] >= '0' && string[i] <= '9') // Check if character is a digit
        {
            int digit = string[i] - '0'; // Convert character to integer
            sum += digit;
            digit_count++;
        }
    }

    printf("%d digits found.\n", digit_count);
    printf("Sum: %d\n", sum);

    return 0;
}

File: exercises/Python/chapter_4 - Strings/1_count_digits.py
"""
1. Count digits in a string

Design a Python program that will ask the user to enter a string.
The program will count the sum of all digits in the string.
"""

# Initialize total and n_digits to 0
total = None
n_digits = 0

# Get a string from the user
string = input("Enter a string of text: ")

# Iterate over the characters in the string
for character in string:
    # If the character is a digit, add its integer value to total
    if character >= "0" and character <= "9":  # character.isdigit()
        # Initialize total
        if total is None:
            total = 0
        total += int(character)
        n_digits += 1

# Output the result
if total == None:
    print("Your string contains no valid digits.")
else:
    print(f"A total of {n_digits} digit(s) found")
    print(f"Sum of digits found in your string is: {total}")
